### PR TITLE
Removing images with multiple tags

### PR DESCRIPTION
**Pull request by ronnielai**
If an image has multiple tags, we need to remove all the tags in order to make docker image removing successful.
#28491

### CODE REVIEW

**Comment by dims** [comment]
Looks good to me 👍

File: pkg/kubelet/dockertools/docker_test.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ import (
 func verifyCalls(t *testing.T, fakeDocker *FakeDockerClient, calls []string) {
 	fakeDocker.Lock()
 	defer fakeDocker.Unlock()
-	verifyStringArrayEquals(t, fakeDocker.called, calls)
+	verifyStringArrayEquals(t, fakeDocker.getCalledNames(), calls)
```
Comment (Random-Liu) [inline pkg/kubelet/dockertools/docker_test.go]: Change this to use `AssertCalls`?

File: pkg/kubelet/dockertools/fake_docker_client.go [hunk 2]
```diff
@@ -468,12 +474,13 @@ func (f *FakeDockerClient) InspectExec(id string) (*dockertypes.ContainerExecIns
...
 
 func (f *FakeDockerClient) ListImages(opts dockertypes.ImageListOptions) ([]dockertypes.Image, error) {
-	f.called = append(f.called, "list_images")
+	f.called = append(f.called, calledDetail{name: "list_images"})
 	err := f.popError("list_images")
 	return f.Images, err
 }
 
 func (f *FakeDockerClient) RemoveImage(image string, opts dockertypes.ImageRemoveOptions) ([]dockertypes.ImageDelete, error) {
+	f.called = append(f.called, calledDetail{name: "remove_image", arguments: []interface{}{image, opts}})
```
Comment (Random-Liu) [inline pkg/kubelet/dockertools/fake_docker_client.go]: Thanks for this! This is quite useful! :)

**Comment by Random-Liu** [comment]
LGTM with nits.

File: pkg/kubelet/dockertools/docker_test.go (hunk 1 above)
Comment (ronnielai) [inline pkg/kubelet/dockertools/docker_test.go]: Done

**Comment by Random-Liu** [comment]
LGTM. Feel free to apply the label after the test passes. :)

**Comment by vishh** [comment]
cc @roberthbailey

**Comment by ronnielai** [comment]
Bumped to P0 to be cherry-picked to 1.3

**Pull request merged**